src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -502,50 +502,86 @@ ExportQueryExecutionTrees::idToLiteral(const Index& index, Id id,
   auto datatype = id.getDatatype();
 
   switch (datatype) {
-    case WordVocabIndex:
-      return ad_utility::triple_component::Literal::literalWithoutQuotes(
-          index.indexToString(id.getWordVocabIndex()));
+    case WordVocabIndex: {
+      auto lit = getLiteralOrIriFromWordVocabIndex(index, id);
+      return getLiteralOrNullopt(lit);
+    }
     case VocabIndex:
     case LocalVocabIndex:
       return handleIriOrLiteral(
           getLiteralOrIriFromVocabIndex(index, id, localVocab),
           onlyReturnLiteralsWithXsdString);
-    case TextRecordIndex:
-      return ad_utility::triple_component::Literal::literalWithoutQuotes(
-          index.getTextExcerpt(id.getTextRecordIndex()));
+    case TextRecordIndex: {
+      auto lit = getLiteralOrIriFromTextRecordIndex(index, id);
+      return getLiteralOrNullopt(lit);
+    }
     default:
       return idToLiteralForEncodedValue(id, onlyReturnLiteralsWithXsdString);
   }
 }
 
+// _____________________________________________________________________________
+std::optional<ad_utility::triple_component::Literal>
+ExportQueryExecutionTrees::getLiteralOrNullopt(
+    std::optional<LiteralOrIri>& litOrIri) {
```

Comment:
This is inefficient.
Get the input as a value (no reference),
and then move the `getLiteral()` out.